### ISSUE TITLE
unable to build 18.09 client on macos

### ISSUE THREAD

**Issue by martenson** [galaxyproject/galaxy#7164]
This is during `make-client`, I tried removing the node_modules/ and also running -production but this happens every build. Same instance builds fine on dev.

```
Traceback (most recent call last):
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/gyp_main.py", line 16, in <module>
    sys.exit(gyp.script_main())
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 545, in script_main
    return main(sys.argv[1:])
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 538, in main
    return gyp_main(args)
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 523, in gyp_main
    generator.GenerateOutput(flat_list, targets, data, params)
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 2170, in GenerateOutput
    part_of_all=qualified_target in needed_targets)
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 795, in Write
    self.Pchify))
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 1190, in WriteSources
    cflags = self.xcode_settings.GetCflags(configname)
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 551, in GetCflags
    archs = self.GetActiveArchs(self.configname)
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 420, in GetActiveArchs
    xcode_archs_default = GetXcodeArchsDefault()
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 118, in GetXcodeArchsDefault
    xcode_version, _ = XcodeVersion()
  File "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 1265, in XcodeVersion
    version = re.match(r'(\d\.\d\.?\d*)', version).groups()[0]
AttributeError: 'NoneType' object has no attribute 'groups'
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Darwin 17.7.0
gyp ERR! command "/usr/local/Cellar/node/11.0.0/bin/node" "/Users/marten/devel/git/galaxy/client/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /Users/marten/devel/git/galaxy/client/node_modules/gulp-sass/node_modules/node-sass
gyp ERR! node -v v11.0.0
```

I also see the following before the error, but that can be just a red herring (https://github.com/nodejs/node-gyp/issues/569):

```
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

**Comment by mvdbeek** [comment]
weird, it's using the global node. It should be the one in your virtualenv. Did you activate the virtualenv before make client ?

**Comment by martenson** [comment]
@mvdbeek I never do that manually, that should happen under the hood even for client build

**Comment by nsoranzo** [comment]
I suppose that we are letting developers shoot themselves in the foot, since `make` it's not run by normal users/deployers.

**Comment by martenson** [comment]
so this is probably an issue with older version of gyp that is used in 18.09 when run without galaxy's `.venv` (even when using the newest node/npm/yarn)

**Comment by dannon** [comment]
Newer node versions (with unfortunate regularity) do break older dependencies.  We saw this with nodejs 10, and again with 11.  That's exactly why the autobuild is locked to a stable, known-compatible node -- this was node 9, until very recently when I relocked it to the 10 LTS after some dependency updates.

And, yeah, it's intentional that we run all of the various exposed client-build targets with whatever node/yarn/etc the developer has active in their environment.  While I like just trusting the developer's intended environment, if folks think we need a little more hand-holding here I wouldn't -1 someone adding sourcing the .venv to the Makefile targets.  I don't think it's a great idea, but personally I usually run the package.json targets directly, bypassing the Makefile, so it wouldn't bother me.
